FinanceManager.UnitOfWork/EntityFramework/UnitOfWork.cs

```diff
@@ -0,0 +1,110 @@
+using System.Collections;
+using FinanceManager.UnitOfWork.Abstructs;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace FinanceManager.UnitOfWork.EntityFramework;
+
+public sealed class UnitOfWork : IUnitOfWork
+{
+    private readonly IServiceProvider _serviceProvider;
+    private Hashtable? _repositories;
+    private DbContext DbContext { get; set; }
+    private IDbContextTransaction? Transaction { get; set; }
+
+    public bool BulkOperation { get; set; }
+
+    internal UnitOfWork(
+        DbContext dbContext,
+        IServiceProvider serviceProvider)
+    {
+        _serviceProvider = serviceProvider;
+        DbContext = dbContext;
+        Transaction = dbContext.Database.BeginTransaction();
+        _repositories = new Hashtable();
+    }
+
+    public T Repository<T>() where T : IUnitOfWorkRepository
+    {
+        var repositoryType = typeof(T);
+        var repositoryTypeName = repositoryType.Name;
+
+        if (_repositories!.ContainsKey(repositoryTypeName))
+        {
+            return (T)_repositories[repositoryTypeName]!;
+        }
+
+        var repository = (T)_serviceProvider.GetService(repositoryType)!;
```

Comment:
Looks like factory candidate, also since your T is strongly typed in context you can enclose this code in a repository factory outside of UOW and even make it strongly typed (registered and resolved directly in DI)